addons/calendar/models/calendar_event.py

```diff
@@ -346,16 +345,44 @@ def _compute_recurrence(self):
             else:
                 event.update(false_values)
 
+    @api.depends('attendee_ids')
+    def _compute_partners(self):
+        for event in self:
+            partner_ids = event.attendee_ids.mapped('partner_id')
+            event.partner_ids = [(6, 0, partner_ids.ids)]
+
+    def _inverse_partners(self):
+        for event in self:
+            attendee_commands = []
+            attendee_partners = event.attendee_ids.mapped('partner_id')
+            new_attendee_partners = event.partner_ids - attendee_partners
+            removed_partners = attendee_partners - event.partner_ids
+            attendees_to_unlink = self.env['calendar.attendee'].search([
+                ('event_id', 'in', self.ids),
+                ('partner_id', 'in', removed_partners.ids),
+            ])
+            attendee_commands += [[2, attendee.id] for attendee in attendees_to_unlink]  # Removes and delete
+            # Don't recreate attendee if partner already attend the event
+            attendee_commands += [
+                [0, 0, dict(partner_id=partner_id.id)]
+                for partner_id in new_attendee_partners
+            ]
+            event.attendee_ids = attendee_commands
+
```

Comment:
Look at the performances PR
https://github.com/odoo/odoo/pull/70648
don't search in a loop, use a defaultdict and do things in batch.